libethereum/Client.cpp

```diff
@@ -670,25 +670,44 @@ void Client::doWork(bool _doWait)
     if (m_syncBlockQueue.compare_exchange_strong(t, false))
         syncBlockQueue();
 
+    if (shouldStop())
+        return;
+
     if (m_needStateReset)
     {
         resetState();
         m_needStateReset = false;
     }
 
+    if (shouldStop())
+        return;
+
+
     t = true;
     bool isSealed = false;
     DEV_READ_GUARDED(x_working)
         isSealed = m_working.isSealed();
     if (!isSealed && !isMajorSyncing() && !m_remoteWorking && m_syncTransactionQueue.compare_exchange_strong(t, false))
         syncTransactionQueue();
 
+    if (shouldStop())
+        return;
+
     tick();
 
+    if (shouldStop())
+        return;
+
     rejigSealing();
 
+    if (shouldStop())
+        return;
+
     callQueuedFunctions();
 
+    if (shouldStop())
```

Comment:
Not sure what these `shouldStop` expressions are doing but it seems it would be better to have it as a PR on its own?

Reply:
I will put small changes from this PR into a separate and then rebase this on top develop. 
this shouldStop() accelerating blockchain reset. Is is a tweak by Andrey. Perhaps better be moved to TestClient.cpp